src/Akka.Persistence.Sql/Db/FooterGenerator.cs

```diff
@@ -0,0 +1,251 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="FooterGenerator.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2023 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using Akka.Persistence.Sql.Config;
+using LinqToDB;
+
+#nullable enable
+namespace Akka.Persistence.Sql.Db
+{
+    public static class FooterGenerator
+    {
+        public static string? GenerateJournalFooter(this JournalConfig config)
+        {
+            var tableName = config.TableConfig.EventJournalTable.Name;
+            var columns = config.TableConfig.EventJournalTable.ColumnNames;
+            var journalFullTableName = string.IsNullOrEmpty(config.TableConfig.SchemaName) 
+                ? tableName : $"{config.TableConfig.SchemaName}.{tableName}";
+			
+            #region SqlServer
+            if (config.ProviderName.StartsWith(ProviderName.SqlServer))
```

Comment:
Ah, is this were secondary indexes are added? Is that not possible to express in the Linq2Db DAO?

Reply:
No sir, Linq2Db does not have any abilities to express any table constraints

Reply:
It would be nice if they can, but its something they would not add any time soon, at least according to my discussion with the dev here: https://github.com/linq2db/linq2db/discussions/3972